prov/shm/src/smr_ep.c

```diff
@@ -1418,6 +1422,43 @@ static void smr_init_sig_handlers(void)
 	pthread_mutex_unlock(&ep_list_lock);
 }
 
+static int smr_create_pools(struct smr_ep *ep, struct fi_info *info)
+{
+	int ret;
+
+	ret = ofi_bufpool_create(&ep->cmd_ctx_pool, sizeof(struct smr_cmd_ctx),
+				 16, 0, info->rx_attr->size,
+				 OFI_BUFPOOL_NO_TRACK);
+	if (ret)
+		goto err;
+
+	ret = ofi_bufpool_grow(ep->cmd_ctx_pool);
+	if (ret)
+		goto free2;
+
+	ret = ofi_bufpool_create(&ep->unexp_buf_pool,
+				 sizeof(struct smr_unexp_buf),
+				 16, 0, 4, OFI_BUFPOOL_NO_TRACK);
```

Comment:
Could you add FI_WARN before goto free? same for the pend buf pool

Reply:
There's already a general warning message for all of the error cases. Would you prefer three different messages?

Reply:
I am ok with having 1 FI_WARN at the bottom, but EP resources may be too general. How about "Unable to allocate buf pools for EP" to match this function's signature?